lib/statsd/instrument/client.rb

```diff
@@ -16,6 +16,9 @@
 #   next major release of this library. While this class may already be functional,
 #   we provide no guarantees about the API and the behavior may change.
 class StatsD::Instrument::Client
+  VOID = Object.new.freeze
```

Comment:
Might be a bit overkill, but maybe a BasicObject subclass would give a better error message:

```ruby
>> DeprectedReturnValue = Class.new(BasicObject)
=> DeprectedReturnValue
>> VOID = DeprectedReturnValue.new
(Object doesn't support #inspect)
=> 
>> VOID.foo
Traceback (most recent call last):
        4: from /opt/rubies/2.6.5/bin/irb:23:in `<main>'
        3: from /opt/rubies/2.6.5/bin/irb:23:in `load'
        2: from /opt/rubies/2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        1: from (irb):3
NoMethodError (undefined method `foo' for #<DeprectedReturnValue:0x00007f82a8968a48>)
```

vs

```ruby
>> Object.new.freeze.plop
Traceback (most recent call last):
        5: from /opt/rubies/2.6.5/bin/irb:23:in `<main>'
        4: from /opt/rubies/2.6.5/bin/irb:23:in `load'
        3: from /opt/rubies/2.6.5/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):4
        1: from (irb):4:in `rescue in irb_binding'
NoMethodError (undefined method `plop' for #<Object:0x00007f82a895b028>)
```

Reply:
I think I will do this because why not? But I am not super worried about people actually using the return value in this way. It's really the `if` case that worries me.